src/parser/shaman/enhancement/modules/talents/LightningShield.tsx

```diff
@@ -7,106 +7,94 @@ import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
 import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
 import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
 import ItemDamageDone from 'interface/ItemDamageDone';
 
+/**
+ * Surround yourself with a shield of lightning for 1 hour.
+ *
+ * Melee attackers have a chance to suffer (3.6% of Attack power)
+ * Nature damage, and add a charge to your shield.
+ * When you Stormstrike, it gains 2 charges.
+ *
+ * At 20 charges, the shield overcharges, causing you to deal an additional
+ * (6% of Attack power) Nature damage with each attack for 10 sec.
+ *
+ * Example Log: https://www.warcraftlogs.com/reports/Qwr8AWzHx6XJfy4p#fight=4&type=auras&ability=192106
+ */
 class LightningShield extends Analyzer {
-  /**
-   * Surround yourself with a shield of lightning for 1 hour.
-   *
-   * Melee attackers have a chance to suffer (3.6% of Attack power)
-   * Nature damage, and add a charge to your shield.
-   * When you Stormstrike, it gains 2 charges.
-   *
-   * At 20 charges, the shield overcharges, causing you to deal an additional
-   * (6% of Attack power) Nature damage with each attack for 10 sec.
-   *
-   * Example Log:
-   */
+  static dependencies = {
+    deathTracker: DeathTracker,
+  };
 
-  protected damage = 0;
-  protected overchargeCount = 0;
+  protected deathTracker!: DeathTracker;
+
+  protected damageGained = 0;
 
   constructor(options: any) {
     super(options);
-    this.active = this.selectedCombatant.hasTalent(SPELLS.LIGHTNING_SHIELD_TALENT.id);
+
+    if (!this.selectedCombatant.hasTalent(SPELLS.LIGHTNING_SHIELD_TALENT.id)) {
+      this.active = false;
+      return;
+    }
 
     this.addEventListener(
       Events.damage.by(SELECTED_PLAYER)
         .spell(SPELLS.LIGHTNING_SHIELD),
-      this.onDamage,
-    );
-
-    this.addEventListener(
-      Events.applybuff.by(SELECTED_PLAYER)
-        .spell(SPELLS.LIGHTNING_SHIELD_TALENT),
-      this.onApplyBuff,
+      this.onLightningShieldDamage,
     );
   }
 
-  onDamage(event: DamageEvent) {
-    this.damage += event.amount + (event.absorbed || 0);
-  }
-
-  onApplyBuff() {
-    this.overchargeCount += 1;
-  }
-
-  get damagePercent() {
-    return this.owner.getPercentageOfTotalDamageDone(this.damage);
+  onLightningShieldDamage(event: DamageEvent) {
+    this.damageGained += event.amount + (event.absorbed || 0);
   }
 
-  get damagePerSecond() {
-    return this.damage / (this.owner.fightDuration / 1000);
-  }
-
-
   get getLightningShieldUptime() {
-    return this.selectedCombatant.getBuffUptime(SPELLS.LIGHTNING_SHIELD_TALENT.id) / this.owner.fightDuration;
+    return this.selectedCombatant.getBuffUptime(SPELLS.LIGHTNING_SHIELD_TALENT.id) / (this.owner.fightDuration - this.deathTracker.totalTimeDead);
   }
 
   get suggestionThresholds() {
     return {
       actual: this.getLightningShieldUptime,
       isLessThan: {
-        major: 1,
+        minor: 1,
+        major: 0.95,
+        recommended: 1.00,
```

Comment:
You dont need to specify recommended in the thresholds. Just list the thresholds (Minor, Average, Major), recommended gets populated with minor automagically.